src/crypto.rs

```diff
@@ -50,36 +56,42 @@ impl DerefMut for MacaroonKey {
     }
 }
 
-impl From<&str> for MacaroonKey {
-    fn from(s: &str) -> Self {
-        generate_derived_key(s.as_bytes())
-    }
-}
-
-impl From<&[u8]> for MacaroonKey {
-    fn from(b: &[u8]) -> Self {
-        generate_derived_key(b)
-    }
-}
-
 impl From<Key> for MacaroonKey {
     fn from(k: Key) -> Self {
         MacaroonKey(k.0)
     }
 }
 
 impl From<[u8; sodiumoxide::crypto::auth::KEYBYTES]> for MacaroonKey {
+    /// Uses bytes directly as a MacaroonKey (with no HMAC)
     fn from(b: [u8; sodiumoxide::crypto::auth::KEYBYTES]) -> Self {
         MacaroonKey(b)
     }
 }
 
 impl From<&[u8; sodiumoxide::crypto::auth::KEYBYTES]> for MacaroonKey {
+    /// Uses bytes directly as a MacaroonKey (with no HMAC)
     fn from(b: &[u8; sodiumoxide::crypto::auth::KEYBYTES]) -> Self {
         MacaroonKey(*b)
     }
 }
 
+impl MacaroonKey {
+    /// Use some seed data to reproducibly generate a MacaroonKey via HMAC.
+    ///
+    /// ```rust
+    /// # use macaroon::MacaroonKey;
+    /// let key = MacaroonKey::generate(b"secret-byte-string");
+    /// let key = MacaroonKey::generate("secret-unicode-string‽".as_bytes());
```

Comment:
Yeah, we don't have something like an `AsBytes` trait bound readily available so think this is totally fine.